bin/backup.sh

```diff
@@ -25,14 +25,47 @@ function gettime() {
   date +%H%M
 }
 
-# Will clear down any previous backups locally (If pushing files elsewhere)
-function cleardown_local() {
-  local yesterday=$(date -d "yesterday" "+${LOCAL_BAK}")
-  local today=$(date "+${LOCAL_BAK}")
-
-  # To make sure we clear up yesterday (if we haven't already)
-  rm -fr ${yesterday}/*.tar.gz
-  rm -fr ${today}/*.tar.gz
+function is_master() {
+  if [[ -d ${ETCD_DATA_DIR}/member ]]; then
+    return 0
+  else
+    return 1
+  fi
+}
+
+function s3_exists() {
+  local s3file="${1}"
+
+  if ! aws s3 ls ${s3file} ; then
+    ec=${PIPESTATUS[*]}
+    if [[ ${ec} -ne 1 ]]; then
+      error_exit "Error detecting s3 file ${s3file}"
+    else
+      # file doesn't exist
+      return 1
+    fi
+  else
+    # file exists in s3
+    return 0
+  fi
+}
+
+function move_s3() {
+  local sourcefile=${1}
+  local destpath=${S3_PATH}${sourcefile#${ETCD_BACKUP_DIR}}
+
+  # Noop when no backup
+  [[ -z ${S3_PATH} ]] && \
+    return 0
+
+  # Check destination before uploading
+  if ! s3_exists ${destpath} ; then
+    aws s3 mv ${sourcefile} ${destpath} --sse aws:kms
```

Comment:
It would seem not. It backs up happily (with the --ses flag). Probably because there is a default KMS ID for the resource or role.